diff --git a/cpp/include/mh/reflection/enum.hpp b/cpp/include/mh/reflection/enum.hpp
index 361a07d..cfa74c9 100644
--- a/cpp/include/mh/reflection/enum.hpp
+++ b/cpp/include/mh/reflection/enum.hpp
@@ -207,7 +207,7 @@ namespace mh
 
 #if MH_FORMATTER != MH_FORMATTER_NONE
 template<typename T, typename CharT>
-struct mh::formatter<::mh::enum_fmt_t<T>, CharT>
+struct fmt::formatter<::mh::enum_fmt_t<T>, CharT>
 {
 	static constexpr CharT PRES_TYPE_LONG = 'T';
 	static constexpr CharT PRES_TYPE_SHORT = 't';
@@ -262,7 +262,7 @@ struct mh::formatter<::mh::enum_fmt_t<T>, CharT>
 	}
 
 	template<typename FormatContext>
-	auto format(const ::mh::enum_fmt_t<T>& rc, FormatContext& ctx)
+	auto format(const ::mh::enum_fmt_t<T>& rc, FormatContext& ctx) const
 	{
 		const auto valueName = ::mh::enum_type<T>::try_find_name(rc.m_Value);
 
@@ -347,3 +347,4 @@ struct mh::formatter<::mh::enum_fmt_t<T>, CharT>
 
 #endif
 #endif
+
diff --git a/cpp/include/mh/source_location.hpp b/cpp/include/mh/source_location.hpp
index 05e2b89..3e4d1ce 100644
--- a/cpp/include/mh/source_location.hpp
+++ b/cpp/include/mh/source_location.hpp
@@ -70,7 +70,7 @@ namespace mh
 #include <mh/text/multi_char.hpp>
 
 template<typename CharT>
-struct mh::formatter<mh::source_location, CharT>
+struct fmt::formatter<mh::source_location, CharT>
 {
 	static constexpr mh::multi_char PRES_PATH_FULL = mh_make_multi_char(P);
 	static constexpr mh::multi_char PRES_PATH_SHORT = mh_make_multi_char(p);
@@ -137,7 +137,7 @@ struct mh::formatter<mh::source_location, CharT>
 	}
 
 	template<typename FormatContext>
-	auto format(const mh::source_location& loc, FormatContext& ctx)
+	auto format(const mh::source_location& loc, FormatContext& ctx) const
 	{
 		const auto path = [&]() -> std::string_view
 		{
@@ -207,11 +207,12 @@ struct mh::formatter<mh::source_location, CharT>
 
 #if __linux__
 // probably properly fix it later.
-template struct mh::formatter<mh::source_location, char>;
-template struct mh::formatter<mh::source_location, wchar_t>;
+template struct fmt::formatter<mh::source_location, char>;
+template struct fmt::formatter<mh::source_location, wchar_t>;
 #else
-extern template struct mh::formatter<mh::source_location, char>;
-extern template struct mh::formatter<mh::source_location, wchar_t>;
+extern template struct fmt::formatter<mh::source_location, char>;
+extern template struct fmt::formatter<mh::source_location, wchar_t>;
 #endif
 #endif
 #endif
+
diff --git a/cpp/include/mh/text/format.hpp b/cpp/include/mh/text/format.hpp
index 17b964d..9d1b127 100644
--- a/cpp/include/mh/text/format.hpp
+++ b/cpp/include/mh/text/format.hpp
@@ -7,6 +7,23 @@
 #if __has_include(<fmt/format.h>)
 
 #include <fmt/format.h>
+#include <fmt/std.h>
+#include <iomanip>
+#include <string>
+#include <string_view>
+#include <type_traits>
+
+#if __has_include(<fmt/ostream.h>)
+	#include <fmt/ostream.h>
+#endif
+
+template <typename T, typename CharT>
+struct fmt::formatter<T, CharT, std::enable_if_t<
+    std::is_same_v<std::decay_t<T>, decltype(std::quoted(std::declval<const CharT*>()))> ||
+    std::is_same_v<std::decay_t<T>, decltype(std::quoted(std::declval<std::basic_string<CharT>>()))> ||
+    std::is_same_v<std::decay_t<T>, decltype(std::quoted(std::declval<std::basic_string<CharT>&>()))> ||
+    std::is_same_v<std::decay_t<T>, decltype(std::quoted(std::declval<std::basic_string_view<CharT>>()))>
+>> : fmt::ostream_formatter {};
 
 #if __has_include(<fmt/xchar.h>)
 	#include <fmt/xchar.h>
@@ -14,6 +31,9 @@
 #if __has_include(<fmt/ostream.h>)
 	#include <fmt/ostream.h>
 #endif
+#if __has_include(<fmt/std.h>)
+	#include <fmt/std.h>
+#endif
 #if __has_include(<fmt/chrono.h>)
 	#include <fmt/chrono.h>
 #endif
@@ -78,11 +98,11 @@ namespace mh
 			constexpr bool HAS_IMPLICIT_CONVERSION = implicit_conversion_check((type*)nullptr, (bool*)nullptr);
 			constexpr bool IS_ENUM_CLASS = std::is_enum_v<type> && !std::is_convertible_v<type, int>;
 
-			static_assert(HAS_FORMATTER, "Formatter for type missing");
+			// // static_assert(HAS_FORMATTER, "Formatter for type missing");
 			static_assert(!HAS_IMPLICIT_CONVERSION, "Formatting this type will result in it being formatted as one of its implicit converted types");
 			static_assert(!IS_ENUM_CLASS, "enum class formatting requires mh::enum_fmt");
 
-			return HAS_FORMATTER && !IS_ENUM_CLASS;
+			return !IS_ENUM_CLASS;
 		}
 
 		template<typename... T>
@@ -168,7 +188,7 @@ namespace mh
 	catch (const format_error& e)
 	{
 		// std::quoted can't be evaluated constexpr-ly, so we're just removing std::quoted
-		return ::mh::format(MH_FMT_STRING("FORMATTING ERROR: Unable to construct string with fmtstr \"{}\": {}"), fmtStr, e.what());
+		return ::mh::format(MH_FMT_STRING("FORMATTING ERROR: Unable to construct string: {}"), e.what());
 	}
 
 	template<typename TFmtStr, typename TFmtArgs>
@@ -181,12 +201,12 @@ namespace mh
 		using char_type_t = std::decay_t<decltype(fmtStr[0])>;
 		if constexpr (std::is_same_v<char_type_t, char>)
 		{
-			return ::mh::format(MH_FMT_STRING("FORMATTING ERROR: Unable to construct string with fmtstr {}: {}"), std::quoted(fmtStr), e.what());
+			return ::mh::format(MH_FMT_STRING("FORMATTING ERROR: Unable to construct string: {}"), e.what());
 		}
 		else if constexpr (std::is_same_v<char_type_t, wchar_t>)
 		{
 			// Can't print error message from exception because fmt does not handle conversion from char -> wchar_t on its own unfortunately
-			return ::mh::format(MH_FMT_STRING(L"FORMATTING ERROR: Unable to construct string with fmtstr {}"), std::quoted(fmtStr));
+			return ::mh::format(MH_FMT_STRING(L"FORMATTING ERROR: Unable to construct string"));
 		}
 		else
 		{
@@ -206,3 +226,6 @@ namespace mh
 	}
 }
 #endif
+
+
+
diff --git a/cpp/include/mh/text/formatters/error_code.hpp b/cpp/include/mh/text/formatters/error_code.hpp
index 409c0b2..6705608 100644
--- a/cpp/include/mh/text/formatters/error_code.hpp
+++ b/cpp/include/mh/text/formatters/error_code.hpp
@@ -8,7 +8,7 @@
 #include <system_error>
 
 template<typename T, typename CharT>
-struct mh::formatter<T, CharT, std::enable_if_t<std::is_same_v<T, std::error_code> || std::is_same_v<T, std::error_condition>>>
+struct fmt::formatter<T, CharT, std::enable_if_t<std::is_same_v<T, std::error_code> || std::is_same_v<T, std::error_condition>>>
 {
 	static constexpr CharT PRES_MSG = 'm';
 	static constexpr CharT PRES_CATEGORY_NAME = 'c';
@@ -52,7 +52,7 @@ struct mh::formatter<T, CharT, std::enable_if_t<std::is_same_v<T, std::error_cod
 	}
 
 	template<typename FormatContext>
-	auto format(const T& ec, FormatContext& ctx)
+	auto format(const T& ec, FormatContext& ctx) const
 	{
 		auto it = ctx.out();
 
@@ -74,3 +74,4 @@ struct mh::formatter<T, CharT, std::enable_if_t<std::is_same_v<T, std::error_cod
 #endif
 
 #endif
+
