diff --git a/cpp/include/mh/reflection/enum.hpp b/cpp/include/mh/reflection/enum.hpp
index 361a07d..bf772dd 100644
--- a/cpp/include/mh/reflection/enum.hpp
+++ b/cpp/include/mh/reflection/enum.hpp
@@ -262,7 +262,7 @@ struct mh::formatter<::mh::enum_fmt_t<T>, CharT>
 	}
 
 	template<typename FormatContext>
-	auto format(const ::mh::enum_fmt_t<T>& rc, FormatContext& ctx)
+	auto format(const ::mh::enum_fmt_t<T>& rc, FormatContext& ctx) const
 	{
 		const auto valueName = ::mh::enum_type<T>::try_find_name(rc.m_Value);
 
diff --git a/cpp/include/mh/source_location.hpp b/cpp/include/mh/source_location.hpp
index 05e2b89..f5a22ec 100644
--- a/cpp/include/mh/source_location.hpp
+++ b/cpp/include/mh/source_location.hpp
@@ -137,7 +137,7 @@ struct mh::formatter<mh::source_location, CharT>
 	}
 
 	template<typename FormatContext>
-	auto format(const mh::source_location& loc, FormatContext& ctx)
+	auto format(const mh::source_location& loc, FormatContext& ctx) const
 	{
 		const auto path = [&]() -> std::string_view
 		{
diff --git a/cpp/include/mh/text/format.hpp b/cpp/include/mh/text/format.hpp
index 17b964d..9a5b130 100644
--- a/cpp/include/mh/text/format.hpp
+++ b/cpp/include/mh/text/format.hpp
@@ -7,6 +7,23 @@
 #if __has_include(<fmt/format.h>)
 
 #include <fmt/format.h>
+#include <fmt/std.h>
+#include <iomanip>
+#include <string>
+#include <string_view>
+#include <type_traits>
+
+#if __has_include(<fmt/ostream.h>)
+	#include <fmt/ostream.h>
+#endif
+
+template <typename T, typename CharT>
+struct fmt::formatter<T, CharT, std::enable_if_t<
+    std::is_same_v<std::decay_t<T>, decltype(std::quoted(std::declval<const CharT*>()))> ||
+    std::is_same_v<std::decay_t<T>, decltype(std::quoted(std::declval<std::basic_string<CharT>>()))> ||
+    std::is_same_v<std::decay_t<T>, decltype(std::quoted(std::declval<std::basic_string<CharT>&>()))> ||
+    std::is_same_v<std::decay_t<T>, decltype(std::quoted(std::declval<std::basic_string_view<CharT>>()))>
+>> : fmt::ostream_formatter {};
 
 #if __has_include(<fmt/xchar.h>)
 	#include <fmt/xchar.h>
@@ -168,7 +185,7 @@ namespace mh
 	catch (const format_error& e)
 	{
 		// std::quoted can't be evaluated constexpr-ly, so we're just removing std::quoted
-		return ::mh::format(MH_FMT_STRING("FORMATTING ERROR: Unable to construct string with fmtstr \"{}\": {}"), fmtStr, e.what());
+		return ::mh::format(MH_FMT_STRING("FORMATTING ERROR: Unable to construct string: {}"), e.what());
 	}
 
 	template<typename TFmtStr, typename TFmtArgs>
@@ -181,12 +198,12 @@ namespace mh
 		using char_type_t = std::decay_t<decltype(fmtStr[0])>;
 		if constexpr (std::is_same_v<char_type_t, char>)
 		{
-			return ::mh::format(MH_FMT_STRING("FORMATTING ERROR: Unable to construct string with fmtstr {}: {}"), std::quoted(fmtStr), e.what());
+			return ::mh::format(MH_FMT_STRING("FORMATTING ERROR: Unable to construct string: {}"), e.what());
 		}
 		else if constexpr (std::is_same_v<char_type_t, wchar_t>)
 		{
 			// Can't print error message from exception because fmt does not handle conversion from char -> wchar_t on its own unfortunately
-			return ::mh::format(MH_FMT_STRING(L"FORMATTING ERROR: Unable to construct string with fmtstr {}"), std::quoted(fmtStr));
+			return ::mh::format(MH_FMT_STRING(L"FORMATTING ERROR: Unable to construct string"));
 		}
 		else
 		{
diff --git a/cpp/include/mh/text/formatters/error_code.hpp b/cpp/include/mh/text/formatters/error_code.hpp
index 409c0b2..5cfc413 100644
--- a/cpp/include/mh/text/formatters/error_code.hpp
+++ b/cpp/include/mh/text/formatters/error_code.hpp
@@ -52,7 +52,7 @@ struct mh::formatter<T, CharT, std::enable_if_t<std::is_same_v<T, std::error_cod
 	}
 
 	template<typename FormatContext>
-	auto format(const T& ec, FormatContext& ctx)
+	auto format(const T& ec, FormatContext& ctx) const
 	{
 		auto it = ctx.out();
 
